src/systems/filecoin_markets/storage_market/storage_market_actor.go

```diff
@@ -0,0 +1,110 @@
+package storage_market
+
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+
+func (sma *StorageMarketActor_I) WithdrawBalance(balance actor.TokenAmount) {
```

Comment:
the user could deposit money, convince a miner to sign a deal and withdraw the money. Am I missing something?

Reply:
if a user withdraws money before the deal hits the chain, then the deal wont be valid.